### PR TITLE
[spark] Add more text predictors

### PR DESCRIPTION
**Pull request by xyang16**
## Description ##

Brief description of what this PR is about

Add more text predictors

### CODE REVIEW

File: extensions/spark/src/main/scala/ai/djl/spark/task/BasePredictor.scala [hunk 1]
```diff
@@ -78,9 +80,13 @@ abstract class BasePredictor[A, B](override val uid: String) extends Transformer
...
   def setTranslator(value: Translator[A, B]): this.type = set(translator, value)
 
+  def setTranslatorFactory(value: TranslatorFactory): this.type = set(translatorFactory, value)
+
+  def setTokenizer(value: String): this.type = set(tokenizer, value)
+
   /** @inheritdoc */
   override def transform(dataset: Dataset[_]): DataFrame = {
-    model = new ModelLoader[A, B]($(engine), $(modelUrl), $(inputClass), $(outputClass))
+    model = new ModelLoader[A, B]($(engine), $(modelUrl), $(inputClass), $(outputClass), $(translator), $(translatorFactory), $(tokenizer))
```
Comment (frankfliu) [inline extensions/spark/src/main/scala/ai/djl/spark/task/BasePredictor.scala:89]: 1. I think Tokenizer should be a member of transaltorFactory.
2. Let's only expose translatorFactory for now, it's a bit confusing have both translator and translatorFactory

File: extensions/spark/src/main/scala/ai/djl/spark/translator/text/QuestionAnsweringTranslator.scala [hunk 2]
```diff
@@ -0,0 +1,49 @@
+///*
```
Comment (lanking520) [inline extensions/spark/src/main/scala/ai/djl/spark/translator/text/QuestionAnsweringTranslator.scala:1]: why we leave a file with full comments?

Comment (xyang16) [reply extensions/spark/src/main/scala/ai/djl/spark/translator/text/QuestionAnsweringTranslator.scala]: Checked in by mistake. Removed

File: extensions/spark/src/main/scala/ai/djl/spark/task/BasePredictor.scala (hunk 1 above)
Comment (xyang16) [reply extensions/spark/src/main/scala/ai/djl/spark/task/BasePredictor.scala]: Changed.

**Review by frankfliu — APPROVED** [review]

**Pull request merged**